src/target_set.c

```diff
@@ -28,6 +28,28 @@
 
 #include "common.h"
 #include "filter_chain.h"
+#include "meta_data.h"
+#include "utils_subst.h"
+
+struct ts_key_list_s
```

Comment:
Please consider using a `char **` instead of this linked list. You can use the convenience functions from `common.h` which should make this quite easy:

```c
char **keys = NULL;
size_t keys_num = 0;

strarray_add (&keys, &keys_num, "foobar");
strarray_free (keys, keys_num);
```